twisted/names/test/test_rootresolve.py

```diff
@@ -209,20 +209,20 @@ def test_lookupChecksClass(self):
         badClass.CLASS = HS
         servers = {
             ('1.1.2.3', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', badClass)],
-                    'authority': [('foo.example.com', Record_NS('ns1.example.com'))],
-                    'additional': [('ns1.example.com', Record_A('10.0.0.2'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', badClass)],
+                    'authority': [(b'foo.example.com', Record_NS(b'ns1.example.com'))],
+                    'additional': [(b'ns1.example.com', Record_A('10.0.0.2'))],
```

Comment:
In my testing, I found that anything which used a domain name as an argument
in the DNS structure required a byte string, and will fail with an exception on Python 3,
but anything passing a string representation of an IP address would not.

So for example, on line 225, this line:

```
         d = resolver.lookupAddress('foo.example.com')
```

fails with:

```
Traceback (most recent call last):
  File "/Users/crodrigues/twisted9/twisted/names/test/test_rootresolve.py", line 225, in test_lookupChecksClass
    d = resolver.lookupAddress('foo.example.com')
  File "/Users/crodrigues/twisted9/twisted/names/common.py", line 81, in lookupAddress
    return self._lookup(name, dns.IN, dns.A, timeout)
  File "/Users/crodrigues/twisted9/twisted/names/root.py", line 126, in _lookup
    self._maximumQueries)
  File "/Users/crodrigues/twisted9/twisted/names/root.py", line 156, in _discoverAuthority
    d = self._query(query, servers, timeout, False)
  File "/Users/crodrigues/twisted9/twisted/names/test/test_rootresolve.py", line 169, in query
    records = server[query.name.name, query.type]
builtins.KeyError: (b'foo.example.com', 1)

twisted.names.test.test_rootresolve.RootResolverTests.test_lookupChecksClass
``

If I changed line 225 to:

```
d = resolver.lookupAddress(b'foo.example.com')
```

would result in this error:

```
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/Users/crodrigues/twisted9/twisted/names/test/test_rootresolve.py", line 225, in test_lookupChecksClass
    d = resolver.lookupAddress(b'foo.example.com')
  File "/Users/crodrigues/twisted9/twisted/names/common.py", line 81, in lookupAddress
    return self._lookup(name, dns.IN, dns.A, timeout)
  File "/Users/crodrigues/twisted9/twisted/names/root.py", line 126, in _lookup
    self._maximumQueries)
  File "/Users/crodrigues/twisted9/twisted/names/root.py", line 156, in _discoverAuthority
    d = self._query(query, servers, timeout, False)
  File "/Users/crodrigues/twisted9/twisted/names/test/test_rootresolve.py", line 169, in query
    records = server[query.name.name, query.type]
builtins.KeyError: (b'foo.example.com', 1)

twisted.names.test.test_rootresolve.RootResolverTests.test_lookupChecksClass
-------------------------------------------------------------------------------
Ran 1 tests in 0.015s

FAILED (errors=1)
```

Changing all the DNS names to byte strings as I did in the patch would
make this test pass on Python 3.